tmt/libraries/file.py

```diff
@@ -0,0 +1,70 @@
+import shutil
+from typing import List, Optional
+
+import tmt
+from tmt.utils import Path
+
+from . import Library, LibraryError, LibraryIdentifierType
+
+
+class File(Library):
+    """
+    Required files
+
+    Takes care of copying required files for specific test or library,
+    more details here:
+    https://tmt.readthedocs.io/en/latest/spec/tests.html#require
+
+    Optional 'parent' object inheriting from tmt.utils.Common can be
+    provided in order to share the cache of already fetched libraries.
+
+    The following attributes are available in the object:
+
+    repo ........ library prefix (git repository name or nick if provided)
+    pattern ..... filename paths and regexes which need to be copied
+    """
+
+    def __init__(
+            self,
+            *,
+            identifier: LibraryIdentifierType,
+            parent: Optional[tmt.utils.Common] = None,
+            logger: tmt.log.Logger,
+            source_location: Path,
+            target_location: Path) -> None:
+        super().__init__(parent=parent, logger=logger)
+
+        self.identifier = identifier
+        self.format: str = identifier.type if (
+            hasattr(identifier, 'type') and isinstance(identifier.type, str)) else 'file'
```

Comment:
How about `self.format = getattr(identifier, 'type', 'file') or 'file'`? If `type` is not present or `None`, you should get the default value.

Reply:
Although, is this even as complicated? IIUIC, `File` instances are always created with `RequireFile` identifier, because of https://github.com/teemtee/tmt/pull/1852/files#diff-438d74f82ead918ec69c4d7945b3d1ce1866548ff61a7584f4b06d581215f054R90, so, how about changing the `identifier` parameter here to respect that? Then it would be sure the `type` attribute exists, there's no chance for `identifier` to be anything but `RequireFile`, and we could simplify: `self.format = identifier.type or 'file'`. Or not?

Reply:
frankly, it already got here to this part, so after a little bit of thought I don't really think any checks are necessary and just setting 'file' should be fine, actually the same is done for libraries anyway